### ISSUE TITLE
Prepare a test framework for ml2cpp

### ISSUE THREAD

**Issue by antoinecarme** [antoinecarme/ml2cpp#1]
We need a way to perform the same checks done for scikit-learn python models for assessing the quality of C++ code generation for machine learning models.

Need (all the code in python):

    1. generate/train a model on a dataset (training = python data.frame)

    2. predict the classes of a test dataset ( test = python data.frame df_out)

    3. generate a C++ code for the model (use the sklearn2sql_heroku for example)

    4. transfer the test dataset to a local CSV file.

    5. Compile and execute the model C++ generated in point 3 ( => predicted CSV file => data.frame cpp_df_out)

    6. compare the two dataframes (df_out and cpp_df_out). A simple merge is OK. The prediction columns should be identical.

In a first time, the goal is just to put in place the framework to be used as a template for coming tests. An xgboost model is OK for this task.

deliverable : A jupyter notebook with all the process.

**Comment by antoinecarme** [comment]
implementation steps, order is not mandatory,  some tasks (C++ design) may survive many steps (see caret2sql implementation): 

https://github.com/antoinecarme/caret2sql/issues?q=is%3Aissue+is%3Aclosed

1. Prototyping / C++ code design / Linear models
2. Decision Tree Models
3. Naive Bayes
4. SVM
5. Neural Nets / MLP
6. Ensemble Models : Random Forests . AdaBoost, Gradient Boosting , ....
7. Transformations / Preprocessing
8. Pipelines / Feature Unions
9. Xgboost / LightGBM
10. Keras / Pytorch
11. Caret
12. Misc Models : feature selection / Earth / ..
13. Web Service
14. Test / Benchmarking

**Comment by antoinecarme** [comment]
Closing
